Riot/Modules/Application/LegacyAppDelegate.m

```diff
@@ -2254,62 +2254,78 @@ - (void)handleAppState
+                return;
+            }
+            
+            if (mainSession.crypto.crossSigning)
+            {
+                // Get the up-to-date cross-signing state
+                MXWeakify(self);
+                [mainSession.crypto.crossSigning refreshStateWithSuccess:^(BOOL stateUpdated) {
+                    MXStrongifyAndReturnIfNil(self);
+                    
+                    MXLogDebug(@"[AppDelegate] handleAppState: crossSigning.state: %@", @(mainSession.crypto.crossSigning.state));
+                    
+                    switch (mainSession.crypto.crossSigning.state)
+                    {
+                        case MXCrossSigningStateCrossSigningExists:
+                            MXLogDebug(@"[AppDelegate] handleAppState: presentVerifyCurrentSessionAlertIfNeededWithSession");
+                            [self.masterTabBarController presentVerifyCurrentSessionAlertIfNeededWithSession:mainSession];
+                            break;
+                        case MXCrossSigningStateCanCrossSign:
+                            MXLogDebug(@"[AppDelegate] handleAppState: presentReviewUnverifiedSessionsAlertIfNeededWithSession");
+                            [self.masterTabBarController presentReviewUnverifiedSessionsAlertIfNeededWithSession:mainSession];
+                            break;
+                        default:
+                            break;
+                    }
+                } failure:^(NSError * _Nonnull error) {
+                    MXLogDebug(@"[AppDelegate] handleAppState: crossSigning.state: %@. Error: %@", @(mainSession.crypto.crossSigning.state), error);
+                }];
+            }
+            
+            // TODO: We should wait that cross-signing screens are done before going further but it seems fine. Those screens
+            // protect each other.
+            
+            // This is the time to check existing requests
+            MXLogDebug(@"[AppDelegate] handleAppState: Check pending verification requests");
+            [self checkPendingRoomKeyRequests];
+            [self checkPendingIncomingKeyVerificationsInSession:mainSession];
+                
+            // TODO: When we will have an application state, we will do all of this in a dedicated initialisation state
+            // For the moment, reuse an existing boolean to avoid register things several times
+            if (!self->incomingKeyVerificationObserver)
+            {
+                MXLogDebug(@"[AppDelegate] handleAppState: Set up observers for the crypto module");
+                
+                // Enable listening of incoming key share requests
+                [self enableRoomKeyRequestObserver:mainSession];
+                
+                // Enable listening of incoming key verification requests
+                [self enableIncomingKeyVerificationObserver:mainSession];
+            }
+        };
+        
+        if (_masterTabBarController.homeViewController.isRoomListDataReady)
```

Comment:
Can we create an async method to put this code?
This will make it easier to customise for forks (some do not have the homeVC)

Reply:
Also i've moved the notification to the RecentsViewController, so it'll be more generic.